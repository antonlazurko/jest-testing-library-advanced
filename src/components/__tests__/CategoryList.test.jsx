import {screen} from '@testing-library/react';
import {CategoryList} from '../CategoryList';
import { renderWithRouter } from '../../utils/testing';

describe('CategoryList', () => {
    test('should render correctly', () => {
        renderWithRouter(<CategoryList catalog={[ {strCategory:"test", idCategory: 'test',
            strCategoryThumb:"/test.png",
            strCategoryDescription: "test"}, {strCategoryDescription: "test", idCategory: 'test2', strCategory: "test", strCategoryThumb: "/test.png"}]}/>);
        expect(screen.getByRole('list')).toMatchSnapshot();
    });
    test('should render correctly with empty array', () => {
        renderWithRouter(<CategoryList/>);
        expect(screen.getByRole('list')).toMatchSnapshot();
    });
})