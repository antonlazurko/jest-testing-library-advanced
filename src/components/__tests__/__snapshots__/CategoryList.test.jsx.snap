// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList should render correctly 1`] = `
<div
  class="list"
  role="list"
>
  <div
    class="card"
    role="article"
  >
    <div
      class="card-image"
    >
      <img
        alt="test"
        src="/test.png"
      />
    </div>
    <div
      class="card-content"
    >
      <span
        class="card-title"
      >
        test
      </span>
      <p>
        test
        ...
      </p>
    </div>
    <div
      class="card-action"
    >
      <a
        class="btn"
        href="/category/test"
      >
        Watch category
      </a>
    </div>
  </div>
  <div
    class="card"
    role="article"
  >
    <div
      class="card-image"
    >
      <img
        alt="test"
        src="/test.png"
      />
    </div>
    <div
      class="card-content"
    >
      <span
        class="card-title"
      >
        test
      </span>
      <p>
        test
        ...
      </p>
    </div>
    <div
      class="card-action"
    >
      <a
        class="btn"
        href="/category/test"
      >
        Watch category
      </a>
    </div>
  </div>
</div>
`;

exports[`CategoryList should render correctly with empty array 1`] = `
<div
  class="list"
  role="list"
/>
`;
